### DISCUSSION TITLE
LSP sometimes says no signature help available where CoC does have it

### DISCUSSION THREAD

**Discussion by cd-a** [LazyVim/LazyVim#2361] [Q&A] [answered]
I used CoC before, and now trying Lazy with the built-in LSP.

I only installed the following extras:
![CleanShot 2024-01-16 at 09 01 33@2x](https://github.com/LazyVim/LazyVim/assets/75008413/1ba44a37-2e5d-4e63-9454-8c8cec0cf2b5)

Not sure if I need none-ls, it makes no difference.

For testing I used a new vite project:
```bash
yarn create vite --template react-ts
cd <folder>
yarn install
vim src/App.tsx
```

Generally, it works fine, but there are some signature helps / hovers that don't work, and this is one example. I hover on the `count` in the JSX

CoC shows signature:
![CleanShot 2024-01-16 at 08 56 06@2x](https://github.com/LazyVim/LazyVim/assets/75008413/47c9d614-71ba-460d-85b2-842858a751f0)

Lazy does not:
![CleanShot 2024-01-16 at 09 05 30@2x](https://github.com/LazyVim/LazyVim/assets/75008413/2d49868c-28a5-44d3-90b8-e322ba64fb71)

However, typing works:
![CleanShot 2024-01-16 at 08 57 05@2x](https://github.com/LazyVim/LazyVim/assets/75008413/b6b49aea-f0cb-4e02-a753-4e67a01cae86)

**edit: same with rust**

```bash
cargo new test
cd test
vim src/main.rs
```

Coc shows signature:
![CleanShot 2024-01-16 at 09 17 47@2x](https://github.com/LazyVim/LazyVim/assets/75008413/e3555168-1757-453f-a7f8-281b9faf3ec6)

Lazy does not:
![CleanShot 2024-01-16 at 09 17 07@2x](https://github.com/LazyVim/LazyVim/assets/75008413/98c47086-e4dd-4386-b70d-074ea7544839)

But again, typing works just fine:
![CleanShot 2024-01-16 at 09 16 51@2x](https://github.com/LazyVim/LazyVim/assets/75008413/bd0d2af2-9732-450a-a261-e478e5806b74)

So the question is, is that just the LSP limitation somehow or a config issue?

**Comment by dpetka2001** [comment, marked as answer]
Are you pressing `K` or `gK`? `vim.lsp.buf.hover` is by default mapped to `K`. That should give you the information you want.

**Reply by cd-a** [reply]
ah damn, I missed that. Thanks, appreciate it. works fine now!
